### PR TITLE
build: simplify and strict tools

### PR DESCRIPTION
**Pull request by viceice**
## Changes:
- make some tools strict
- unify schema generation, so only one generation code exists.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by rarkins** [comment]
Need to double check that it creates the same schema. I thought I saw some small changes in logic between the two scripts last time

**Comment by viceice** [comment]
I've generated with both and got 100% same output

File: tools/docs/schema.ts [hunk 1]
```diff
@@ -22,7 +22,7 @@ options.sort((a, b) => {
 const properties = schema.properties as Record<string, any>;
 
 function createSingleConfig(option: RenovateOptions): Record<string, unknown> {
-  const temp = {} as Record<string, any> & RenovateOptions;
+  const temp = {} as Record<string, any> & Partial<RenovateOptions>;
```
Comment (JamieMagee) [inline tools/docs/schema.ts:25]:
```suggestion
  const temp: Record<string, any> & Partial<RenovateOptions> = {};
```

Comment (viceice) [reply tools/docs/schema.ts:25]: Makes no difference, but ok. 😏

**Review by rarkins — APPROVED** [review]

**Pull request merged**